lib/extensions/permissions.js

```diff
@@ -215,16 +220,74 @@ async function setAccess (bundleId, permissionsMapping) {
     const permissionsArg = _.toPairs(wixPermissions)
       .map((x) => `${x[0]}=${formatStatus(x[1])}`)
       .join(',');
-    await execWix.bind(this)([
+    const execWixFn = async () => await execWix.bind(this)([
       '--byId', this.udid,
       '--bundle', bundleId,
       '--setPermissions', permissionsArg,
     ]);
+    const shouldWaitForSpringboardRestart = !_.isEmpty(
+      _.intersection(SERVICES_NEED_SPRINGBOARD_RESTART, _.keys(wixPermissions))
+    );
+    if (shouldWaitForSpringboardRestart) {
+      const [didTimeout] = await runAndWaitForSpringboardRestart.bind(this)(
+        execWixFn, SPRINGBOARD_RESTART_TIMEOUT_MS
+      );
+      if (didTimeout) {
+        this.log.warn(
+          `The Springboard service did not restart after ${SPRINGBOARD_RESTART_TIMEOUT_MS}ms ` +
+          `timeout. This might lead to unexpected consequences later.`
+        );
+      }
+    } else {
+      await execWixFn();
+    }
   }
 
   return true;
 }
 
+/**
+ *
+ * @this {import('../types').CoreSimulator}
+ * @template {any} T
+ * @param {() => Promise<T>} fn
+ * @param {number} timeoutMs
+ * @returns {Promise<[boolean, T]>}
+ */
+async function runAndWaitForSpringboardRestart(fn, timeoutMs) {
```

Comment:
the name of this function might be confusing as it also restarts the spotlight service.

Perhaps, we could rename it to something more specific, like runAndWaitForSystemReadiness

Reply:
Perhaps good naming.

The springboard restart caused applications' restarts, such as `Spotlight.app`. So, I initially thought such application restarts can also be part of the springboard restart method. Then, this method is ready == other services/apps restarted by springboard are also ready.